bundles/alpha.codegen/src/alpha/codegen/writeC/ReduceExprConverter.xtend

```diff
@@ -0,0 +1,224 @@
+package alpha.codegen.writeC
+
+import alpha.codegen.Expression
+import alpha.codegen.Factory
+import alpha.codegen.FunctionBuilder
+import alpha.codegen.MacroStmt
+import alpha.codegen.ProgramBuilder
+import alpha.codegen.isl.ASTConverter
+import alpha.codegen.isl.LoopGenerator
+import alpha.model.ReduceExpression
+import fr.irisa.cairn.jnimap.isl.ISLAff
+import fr.irisa.cairn.jnimap.isl.ISLConstraint
+import fr.irisa.cairn.jnimap.isl.ISLDimType
+import fr.irisa.cairn.jnimap.isl.ISLSpace
+import java.util.HashSet
+
+/**
+ * Converts an Alpha reduce expression into the appropriate C AST nodes.
+ * A new function is created and added to the program which computes the reduction,
+ * and the appropriate function call expression is returned.
+ */
+class ReduceExprConverter {
```

Comment:
This class is what handles reduce expressions, so should be looked at more carefully as well.

Reply:
This looks mostly good as well! I just left a few comments related to duplicate naming issues.